pkg/action/cl001/ac009/executor.go

```diff
@@ -2,8 +2,151 @@ package ac009
 
 import (
 	"context"
+
+	infrastructurev1alpha2 "github.com/giantswarm/apiextensions/v2/pkg/apis/infrastructure/v1alpha2"
+	"github.com/giantswarm/k8sclient/v4/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	valuemodifierpath "github.com/giantswarm/valuemodifier/path"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	k8sruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/awscnfm/v12/pkg/client"
+	"github.com/giantswarm/awscnfm/v12/pkg/env"
+	"github.com/giantswarm/awscnfm/v12/pkg/label"
+)
+
+const (
+	kubeSystemNamespace = "kube-system"
+
+	draughtsmanNamespace                  = "draughtsman"
+	draughtsmanConfigMapName              = "draughtsman-values-configmap"
+	draughtsmanConfigMapDataKey           = "values"
+	draughtsmanConfigMapDockerRegistryKey = "Installation.V1.Registry.Domain"
 )
 
 func (e *Executor) execute(ctx context.Context) error {
+	var err error
+
+	var cpClients k8sclient.Interface
+	{
+		c := client.ControlPlaneConfig{
+			Logger: e.logger,
+
+			KubeConfig: env.KubeConfig(),
+		}
+
+		cpClients, err = client.NewControlPlane(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var tcClients k8sclient.Interface
+	{
+		c := client.TenantClusterConfig{
+			ControlPlane: cpClients,
+			Logger:       e.logger,
+
+			Scope: e.scope,
+		}
+
+		tcClients, err = client.NewTenantCluster(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	// awsRegion is necessary to execute aws-cli call
+	var awsRegion string
+	{
+		var cr infrastructurev1alpha2.AWSCluster
+		{
+			var list infrastructurev1alpha2.AWSClusterList
+			err := cpClients.CtrlClient().List(
+				ctx,
+				&list,
+				k8sruntimeclient.InNamespace(cr.GetNamespace()),
+				k8sruntimeclient.MatchingLabels{label.Cluster: e.tenantCluster},
+			)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			if len(list.Items) == 0 {
+				return microerror.Mask(notFoundError)
+			}
+			if len(list.Items) > 1 {
+				return microerror.Mask(tooManyCRsError)
+			}
+
+			cr = list.Items[0]
+		}
+
+		awsRegion = cr.Spec.Provider.Region
+	}
+
+	// dockerRegistry is needed in order to spawn pod with proper docker image that will execute aws-cli call
+	var dockerRegistry string
+	{
+		dockerRegistry, err = e.fetchDockerRegistry(ctx, cpClients.K8sClient())
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	err = e.testAWSApiCalls(ctx, tcClients.K8sClient(), awsRegion, dockerRegistry)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// testAWSApiCalls will spawn a job in k8s tenant cluster to test calling AWS API to ensure kiam works as expected
+func (e *Executor) testAWSApiCalls(ctx context.Context, tcClient kubernetes.Interface, awsRegion string, dockerRegistry string) error {
+	networkPolicy := jobNetworkPolicy()
+	_, err := tcClient.NetworkingV1().NetworkPolicies(kubeSystemNamespace).Create(ctx, networkPolicy, metav1.CreateOptions{})
```

Comment:
When we create the network policies here, will we take care of cleaning them up in the next steps? It would be cool to ensure idempotence in our actions.

Reply:
there will have to be another action for deleting probabbly ( we could do it in the action where we check results but that might be not the right design)